src/control/tools/EditSelection.cpp

```diff
@@ -639,6 +641,43 @@ CursorSelectionType EditSelection::getSelectionTypeForPos(double x, double y, do
 	return CURSOR_SELECTION_NONE;
 }
 
+void EditSelection::snapRotation()
+{
+	double epsilon = 0.1;
+	if(std::abs(this->rotation) < epsilon)
```

Comment:
do this values to a const double, and then do a loop, like:

const double ROTATION_LOCK = {0, M_PI / 2.0, M_PI ....};
....
for (int i = 0; i < sizeof(ROTATION_LOCK) / siezof(ROTATION_LOCK[0]); i++)
{
   if (std::abs(ROTATION_LOCK[i]) < epsilon)
      this->rotation = ROTATION_LOCK[i];
}

(yes, this formatting is not OK ;-))

Reply:
Hold down keys (CTRL, ALT, SHIFT) are within the mouse event, I did it wrong on my last change. You don't need key listing.

Reply:
> Hold down keys (CTRL, ALT, SHIFT) are within the mouse event, I did it wrong on my last change. You don't need key listing.

I don't get what your're talking about here